File: tests/bench/closure_capture.lua
local makers = {}
for i = 0, 9999 do
    local x = i
    local y = i * 2
    makers[i] = function() return x + y end
end
local sum = 0
for i = 0, 9999 do
    sum = sum + makers[i]()
end
print(sum)

File: tests/bench/node/closure_capture.js
function makeAdder(x, y) { return function() { return x + y; }; }
let makers = [];
for (let i = 0; i < 10000; i++) makers.push(makeAdder(i, i * 2));
let sum = 0;
for (let i = 0; i < 10000; i++) sum += makers[i]();
console.log(sum);
